tools/jobs.php

```diff
@@ -292,6 +296,52 @@ protected function list(): void
             );
     }
 
+    /**
+     * Dispatch jobs into the queue
+     */
+    protected function run(): void
+    {
+        $jobQueue = app('pkpJobQueue');
+        
+        $queue = $this->getParameterList()['queue'] ?? Config::getVar('queues', 'default_queue', 'queue');
+
+        if ( in_array('--test', $this->getParameterList()) ) {
+
+            $queue = PKPJobModel::TESTING_QUEUE;
+        }
+
+        if ( $queue && is_string($queue) ) {
+
+            $jobQueue = $jobQueue->forQueue($queue);
+        }
+
+        $jobBuilder = $jobQueue->getJobModelBuilder();
+
+        if ( ($jobCount = $jobBuilder->count()) <= 0 ) {
+            
+            $this->getCommandInterface()->getOutput()->success(
+                __(
+                    'admin.cli.tool.jobs.available.options.run.empty.description', 
+                    ['queueName' => $queue,]
+                )
+            );
+
+            return;
+        }
+
+        while($jobBuilder->count()) {
```

Comment:
Just checking -- is this resilient against failed jobs? (i.e. if a job remains in the queue because it failed, will this hang?)

Reply:
@asmecher  No that is how it's works . basically a job will and should go (unless there is an issue with our implementation or design), it will go to failed table after 3 failed attempt. the implementation for cron or schedular do utilise laravel's worker but not through a `daemon` . so we need a way to pop and push jobs into the worker queue one by one through the `Illuminate\Queue\Worker::runNextJob()` . Also the DB query check of max attempt as a precaution to make sure if a job has been attempted max time and yet present in the `jobs` table , will not be picked up for processing . Again, a job that is reserved by marking the `reserved_at` column will not be picked up so we should be safe from a loop running for infinite time. But we can make it better by making a `max job processing limit in a single call` just to be safe . I will do that just to have an infinite time running trap handler but that can not be implemented for `daemon` feature as that is the purpose of `daemon` feature .

Reply:
Thanks, that sounds good!